packages/headless/src/controllers/commerce/context/cart/headless-cart.ts

```diff
@@ -0,0 +1,131 @@
+import {validateOptions} from '../../../../utils/validate-payload';
+import {CommerceEngine} from '../../../../app/commerce-engine/commerce-engine';
+import {
+  buildController,
+  Controller,
+} from '../../../controller/headless-controller';
+import {loadReducerError} from '../../../../utils/errors';
+import {cartReducer as cart} from '../../../../features/commerce/context/cart/cart-slice';
+import {
+  addItem,
+  removeItem,
+  setItems,
+  updateItemQuantity,
+} from '../../../../features/commerce/context/cart/cart-actions';
+import {cartSchema} from '../../../../features/commerce/context/cart/cart-validation';
+
+export interface CartOptions {
+  items?: CartItem[];
+}
+
+export interface CartItem {
+  productId: string;
+  quantity: number;
+}
+
+export interface CartProps {
+  /**
+   * The initial options that should be applied to this `Cart` controller.
+   */
+  options?: CartOptions;
+}
+
+/**
+ * The `Cart` controller exposes methods for managing the cart in a commerce interface.
+ */
+export interface Cart extends Controller {
+  /**
+   * Sets the cart items.
+   * @param cart - The new cart.
+   */
+  setItems(cart: CartItem[]): void;
+
+  /**
+   * Adds a cart item.
+   * @param item - The new cart item.
+   */
+  addItem(item: CartItem): void;
+
+  /**
+   * Removes a cart item.
+   * @param productId - The cart item's product id to remove
+   */
+  removeItem(productId: string): void;
```

Comment:
If a user has added, say, 3 of a single item to their card, will it remove _**all**_ of that item? Or will it only remove 1 at a time?

Based on the fact that `quantity` is a part of the `CartItem` interface, and there's a `updateItemQuantity` method below, I'll assume it's the former. That being said, it may be worthwhile to clarify in the description.

Reply:
It will remove **all** of that item! Since the method exposes only a `productId`, it seems clear enough to me that all of that item's quantity will be removed. Should that be made more explicit through a docstring? Something like this?
```
* Remove all of an item's quantity
```

See: https://github.com/coveo/ui-kit/pull/3190/files#diff-c81cbfcfb6410f7305a9ccc75ecf5f40ff139616b69386fde2ff97ee1d532ec6R38-R43

Reply:
Thanks for clarifying! Honestly, now that you mention it, it's not confusing once you realize that both the `productId` _**and**_ `quantity` are bound to the same `CartItem` object. The more I reviewed this PR, the more obvious it became, so the original description is likely enough for implementers to understand what's going on 😅